scripts/determine-sf-version.js

```diff
@@ -0,0 +1,33 @@
+const pjson = require('../package.json');
+const shelljs = require('shelljs');
+
+const getVersion = () => {
+  const sfdxVersion = pjson.version;
+
+  // Get all npm dist-tags for sfdx-cli
+  const distTags = JSON.parse(shelljs.exec('npm view sfdx-cli dist-tags --json', { silent: true }).stdout.trim());
+
+  // Find the dist-tag that matches the sfdx-cli package.json version
+  const distTag = Object.entries(distTags).find(([, version]) => version === sfdxVersion)[0];
+
+  // Now get the version of @salesforce/cli for the same dist-tag
+  let sfVersion = JSON.parse(
+    shelljs.exec(`npm view @salesforce/cli@${distTag} version --json`, { silent: true }).stdout.trim()
+  );
+
+  // Make sure that the version of @salesforce/cli does not start with 2
+  if (sfVersion.startsWith('2')) {
```

Comment:
seems like `sfVersion` should be typed as string here to support comment above

Reply:
These scripts are just js files so I cannot add types. Sorry if I'm missing your meaning?